index.js

```diff
@@ -36,13 +78,32 @@ module.exports = function (opts = {}) {
     }
 
     app.authboot = {};
+
+    const concurrency = opts.maxAuthConcurrency
+      || options.maxAuthConcurrency
+      || app.config.get('auth:maxAuthConcurrency')
+      || 3;
```

Comment:
is this safe at a default? I don't have a good sense of how fast the callback queues will get flushed. like what requests per second will start causing failures by the guesses.size being > concurrency? @bmeck

Reply:
this is 3 concurrent guesses on a single user. not 3 overall checks.

Reply:
@bmeck I agree but if most of our requests are coming from a single user at about 1.5k requests per minute, will this value be sufficient? Or whats a good way to gauge what to make this value

Reply:
single user/guess pair gets coalesced to single check with burst of 1 full loop. this value is purely for people doing multiple auths on different name/guess pairs, it should be quite small normally and could even be `1` in the usual case.

Reply:
Understood, I was misinterpreting guesses.size which is actually referencing different password and not the size of the callback queue.